patches/chromium/feat_add_storage_quota_capability_for_partitions.patch

```diff
@@ -0,0 +1,232 @@
+   const auto total = quotaAvailability.total;
+   const auto available = quotaAvailability.available;
+ 
+@@ -2955,9 +2967,15 @@ QuotaAvailability QuotaManagerImpl::CallGetVolumeInfo(
+ }
+ 
+ // static
+-QuotaAvailability QuotaManagerImpl::GetVolumeInfo(const base::FilePath& path) {
+-  return QuotaAvailability(base::SysInfo::AmountOfTotalDiskSpace(path),
+-                           base::SysInfo::AmountOfFreeDiskSpace(path));
++QuotaAvailability QuotaManagerImpl::GetVolumeInfo(const base::FilePath& path, const int64_t & quota_size) {
++  if(!quota_size) {
++    return QuotaAvailability( base::SysInfo::AmountOfTotalDiskSpace(path),
++                            base::SysInfo::AmountOfFreeDiskSpace(path));
++  }
++  else {
++    return QuotaAvailability( base::SysInfo::AmountOfTotalDiskSpace(path),
++                            quota_size);
++  }
+ }
+ 
+ }  // namespace storage
+diff --git a/storage/browser/quota/quota_manager_impl.h b/storage/browser/quota/quota_manager_impl.h
+index 6941faf7f4ab294bc48618d1b9b45238c9a6e9a6..086c6152c08e4a97c8bd90b58eaf746355b401de 100644
+--- a/storage/browser/quota/quota_manager_impl.h
++++ b/storage/browser/quota/quota_manager_impl.h
+@@ -154,7 +154,7 @@ class COMPONENT_EXPORT(STORAGE_BROWSER) QuotaManagerImpl
+   // Function pointer type used to store the function which returns
+   // information about the volume containing the given FilePath.
+   // The value returned is the QuotaAvailability struct.
+-  using GetVolumeInfoFn = QuotaAvailability (*)(const base::FilePath&);
++  using GetVolumeInfoFn = QuotaAvailability (*)(const base::FilePath&,const int64_t &);
+ 
+   static constexpr int64_t kGBytes = 1024 * 1024 * 1024;
+   static constexpr int64_t kNoLimit = INT64_MAX;
+@@ -468,6 +468,7 @@ class COMPONENT_EXPORT(STORAGE_BROWSER) QuotaManagerImpl
+     eviction_disabled_ = disable;
+   }
+ 
++ void  SetStartupQuota(const int64_t start_quota);
+   // Testing support for handling corruption in the underlying database.
+   //
+   // Runs `corrupter` on the same sequence used to do database I/O,
+@@ -730,8 +731,10 @@ class COMPONENT_EXPORT(STORAGE_BROWSER) QuotaManagerImpl
+       bool is_bootstrap_task = false);
+ 
+   static QuotaAvailability CallGetVolumeInfo(GetVolumeInfoFn get_volume_info_fn,
+-                                             const base::FilePath& path);
+-  static QuotaAvailability GetVolumeInfo(const base::FilePath& path);
++                                             const base::FilePath& path,
++                                             const int64_t& quota_size);
++  static QuotaAvailability GetVolumeInfo(const base::FilePath& path,
++                                            const int64_t& quota_size);
+ 
+   const bool is_incognito_;
+   const base::FilePath profile_path_;
+@@ -828,6 +831,8 @@ class COMPONENT_EXPORT(STORAGE_BROWSER) QuotaManagerImpl
+   // QuotaManagerImpl::GetVolumeInfo.
+   GetVolumeInfoFn get_volume_info_fn_;
+ 
++  int64_t start_quota_ = 0;
```

Comment:
Can you make this absl::optional<int64_t>

Reply:
Done.